src/tokens.ts

```diff
@@ -0,0 +1,16 @@
+import { Token } from '@lumino/coreutils';
+
+namespace SchedulerExtension {
+  export interface ICustomEnvironmentProps {
+    state: any;
+    setState: any;
+  }
+
+  export type ICustomEnvironment = React.FC<ICustomEnvironmentProps>;
+
+  export const ICustomEnvironment = new Token<ICustomEnvironment>(
+    'scheduler-extension-custom-environment:ICustomEnvironment'
```

Comment:
I believe the convention for token id is 'namespace/extension:component'. Use `@jupyterlab/scheduler:ICustomEnvironment` here.